src/components/Content/GenericAction.vue

```diff
@@ -113,6 +114,55 @@ export default {
       if(this.action.requiredUpgrade && !this.$store.getters['upgrades/get'](this.action.requiredUpgrade)) return true;
       return false;
     },
+    uncompleted() { // find items in output and return true if any of them are absent from our completion page
+      if(!this.$store.getters['settings/showCompletionLines']) return false;
+      if(this.action.item){
+        var itemCount = this.$store.getters["completion/getItem"](this.action.item) || 0;
+        if(!itemCount){
+          //console.log("Not enough of "+this.action.item+"! Returning true (1)...");
+          return true; 
+        }
+      } else if(this.action.items){
+        var itemCount = this.$store.getters["completion/getItem"](this.action.items.id) || 0;
+        if(!itemCount){
+          //console.log("Not enough of "+this.action.items.id+"! Returning true (2)...");
+          return true; 
+        }
+      } else if(this.action.itemTable){
+        for (let i = 0; i < this.action.itemTable.length; i++){
+          var itemCount = this.$store.getters["completion/getItem"](this.action.itemTable[i].id) || 0;
+          if(!itemCount){
+            //console.log("Not enough of "+this.action.itemTable[i].id+"! Returning true (3)...");
+            return true; 
+          }
+        }
+      } else if(this.action.itemTables){
+        for (let tables of this.action.itemTables){
+          if(tables.itemTable){

```

Comment:
It might be better to do this with a bit of recursion instead of checking for `itemTable`, `items`, and `item` again.  Both for keeping things clean, and also because I think it's technically possible to have `itemTables` within `itemTables` (though I don't remember if this actually happens in practice).

You can refer to `itemChanceUtils.js` for a general pattern you could follow for that. It might also be a good idea to move some of that logic over there just so this file doesn't get too bloated, but I wouldn't require this.

Reply:
> itemTables within itemTables

Fortunately I don't think this occurs anywhere. I did see the implementation in itemChanceUtils.js when I was about halfway through writing this, but it didn't occur to me to rewrite what I had... 🤦 I think you're right that that implementation would be better

As for performance, I was surprised to not see any noticeable drop when I played with it for a few hours last night, but I did have the same concern as you. I would bet that if you were on a device that was just barely able to run the game before then this would make it worse, hence making it an opt-in setting